include/benchmark/benchmark.h

```diff
@@ -866,7 +890,8 @@ class BENCHMARK_EXPORT State {
  private:
   bool started_;
   bool finished_;
-  bool error_occurred_;
+  bool skipped_;
+  bool skip_is_error_;
```

Comment:
my first draft kept the name(s) as `error_ocurred_` but then i had a bunch of `if skipped | error` statements.  it was much cleaner when i refactored to make `error` a subset of `skipped`

Reply:
do you think this would be cleaner as an enum? 

```
enum Skipped {
  NotSkipped,
  SkippedWithMessage,
  SkippedWithError,
};
```

or does it just make things more complex?

Reply:
the way it is is slightly fragile if someone sets "skip_is_error_" without setting "skipped_".

Reply:
i agree.  i will change it to an enum.